src/main/resources/schema.graphql

```diff
@@ -149,8 +152,42 @@ type SampleMatchedAnalysisPair {
     tumourSampleAnalysis: Analysis
 }
 
+enum AnalysisSortField {
+    analysisId,
+    analysisState,
+    publishedAt,
+    updatedAt,
+    firstPublishedAt
+}
+
+enum SortOrder {
+    asc,
+    desc
+}
+
+input AnalysisSort {
+    fieldName: AnalysisSortField!
+    order: SortOrder!
+}
+
+type SearchResultInfo {
+    contentCount: String!
+    hasNextFrom: String!
+    totalHits: String!
+}
+
+type AnalysesSearchResult {
+    content: [Analysis!]
+    info: SearchResultInfo!
+}
+
+type AggregationResult {
+    totalHits: String!
+}
+
 extend type Query {
-    analyses(filter: AnalysisFilter, page: Page): [Analysis]
+    analyses(filter: AnalysisFilter, page: Page, sorts: [AnalysisSort!]): AnalysesSearchResult!
+    aggregateAnalyses(filter: AnalysisFilter): AggregationResult!
     files(filter: FileFilter, page: Page): [File]
```

Comment:
I know we don't currently use the file search much but how much work would it be to roll out the same changes for this as well?

Reply:
haha actually yea I'm doing that now, was trying to get it in before you guys had a chance to review :stuck_out_tongue: .. The only thing I'm still trying to figure out is what the default sort should be for files...

Reply:
I'll open in a new PR.. when its ready now :+1: